executor/src/witgen/machines/double_sorted_witness_machine.rs

```diff
@@ -17,23 +17,34 @@ use powdr_ast::analyzed::{
     AlgebraicExpression as Expression, AlgebraicReference, Identity, IdentityKind, PolyID,
 };
 
-const EXPECTED_WITNESSES: [&str; 7] = [
+/// If all witnesses of a machine have a name in this list (disregarding the namespace),
+/// we'll consider it to be a double-sorted machine.
+/// This does not include the selectors, which are dynamically added to this list.
+const ALLOWED_WITNESSES: [&str; 8] = [
     "m_value",
     "m_addr",
     "m_step",
     "m_change",
     "m_is_write",
-    "m_is_read",
     "m_is_bootloader_write",
+    "m_diff_upper",
+    "m_diff_lower",
 ];
 
 const DIFF_COLUMNS: [&str; 2] = ["m_diff_upper", "m_diff_lower"];
```

Comment:
is this still needed then?

Reply:
Yes, it is used to set `has_diff_columns` in line 126. I could remove them from `ALLOWED_WITNESSES` and chain the diff columns, should I do that? Not sure what to call `ALLOWED_WITNESSES` then, because it wouldn't contain all allowed witnesses anymore.

Reply:
fine as is